addons/map/functions/fnc_switchFlashlight.sqf

```diff
@@ -16,10 +16,18 @@
 
 #include "script_component.hpp"
 
-params ["_flashlight"];
+params ["_newFlashlight"];
+private _oldFlashlight = GVAR(flashlightInUse);
```

Comment:
There's three "levels of detail" you can adopt when doing this kind of stuff:

A - Using a GVAR. State is shared between all local units. If you remote control a unit using zeus, team switch, etc the flashlight state carries over. No network traffic at all.

B - Using `_unit setVariable`, but privately. The state is unique to each unit, so it supports team switch, remote control, etc but only for local units. If you e.g. remote control a remote unit, the correct state won't be displayed. No network traffic at all.

C - Using `_unit setVariable` publicly. The state is unique to each unit, even if they are remote, locality changes, etc. Requires network traffic on each change of state.

Currently this uses approach A. I assume @jonpas is refering to B and @voiperr is talking about C. I consider C the ideal for this, but it's hardly worth the effort.